debian/patches/no-nocloud-network.patch

```diff
@@ -0,0 +1,124 @@
+Description: Don't allow network-config
+This may add a new wait time for a file that doesn't exist on existing series
+so patch it out.
+
+Author: Brett Holman <brett.holman@canonical.com>
+Last-Update: 2024-08-02
+
+--- a/cloudinit/sources/DataSourceNoCloud.py
++++ b/cloudinit/sources/DataSourceNoCloud.py
+@@ -199,7 +199,7 @@ class DataSourceNoCloud(sources.DataSour
+             )
+             mydata["user-data"] = ud
+             mydata["vendor-data"] = vd
+-            mydata["network-config"] = network
++            mydata["network-config"] = None
```

Comment:
In noble, this hunk is different. There we just don't set `mydata["network-config"]`:
```diff
@@ -199,7 +199,6 @@
             )
             mydata["user-data"] = ud
             mydata["vendor-data"] = vd
-            mydata["network-config"] = network
             found.append(seedfrom)

         # Now that we have exhausted any other places merge in the defaults
```

Reply:
And this patch is also missing this other hunk from noble:
```
--- a/cloudinit/sources/DataSourceNoCloud.py
+++ b/cloudinit/sources/DataSourceNoCloud.py
@@ -190,7 +190,7 @@

             # This could throw errors, but the user told us to do it
             # so if errors are raised, let them raise
-            md_seed, ud, vd, network = util.read_seeded(seedfrom, timeout=None)
+            md_seed, ud, vd, _ = util.read_seeded(seedfrom, timeout=None)
             LOG.debug("Using seeded cache data from %s", seedfrom)

             # Values in the command line override those from the seed
```
This one isn't that important, but the previous one seems a bigger difference.

Reply:
Those hunks are roughly inert in either case because mydata is initialized as `"network-config": None`, but it's better not to have a quilt patch diff that we have to look at across series and I had refreshed one patch and not refreshed the other completely. Thanks for pointing it out, I'll bring jammy into the fold w/ the same quilt patch.